src/App.tsx

```diff
@@ -2,108 +1,71 @@
+import React, { useState } from 'react';
+
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const optionsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(optionsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const firstVisibleItemIndex = (currentPage - 1) * perPage;
+  const lastVisibleItemIndex = Math.min(items.length, currentPage * perPage);
+
+  const visibleItems = items.slice(firstVisibleItemIndex, lastVisibleItemIndex);
+
+  const handlePageChange = (newPage: number) => {
+    setCurrentPage(newPage);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
```

Comment:
The `setCurrentPage(1)` ensures that the user is taken back to the first page after changing the number of items per page. This is a good user experience practice as it prevents users from landing on an empty or non-existent page.